dace/sdfg/validation.py

```diff
@@ -191,6 +191,18 @@ def validate_state(state: 'dace.sdfg.SDFGState',
         except Exception as ex:
             raise InvalidSDFGNodeError("Node validation failed: " + str(ex), sdfg, state_id, nid) from ex
 
+        if Config.get_bool('check_race_conditions'):
+            if isinstance(node, nd.AccessNode):
+                incoming_edge_memlet_ranges = []
+                for e in state.in_edges(node):
+                    incoming_edge_memlet_ranges.append(e.data.subset)
+                if (len(incoming_edge_memlet_ranges) > 0):
+                    sorted(incoming_edge_memlet_ranges)
```

Comment:
what does this do? If you meant to sort the memlets, `sorted` is a generator so this doesn't actually sort the list in-place.

Reply:
additionally, this may raise an exception if two incomparable symbolic expressions are used (for example: 0:N and 0:M). I think it results in a `TypeError` but please double-check. If we cannot compare memlets, then we should skip this check.

Reply:
I tried using `.sort()` method to address the first issue.
I have also added an example for depicting what happens when we use symbolic ranges. Please have a look at it and share your opinion.